apps/src/templates/SwipePrompt.jsx

```diff
@@ -0,0 +1,138 @@
+/* global navigator */
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import cookies from 'js-cookie';
+import {dismissSwipeOverlay} from '@cdo/apps/templates/arrowDisplayRedux';
+import trackEvent from '@cdo/apps/util/trackEvent';
+
+const styles = {
+  overlay: {
+    position: 'absolute',
+    zIndex: 1,
+    opacity: '90%'
+  },
+  minecraft: {
+    top: '62px',
+    left: '17px'
+  }
+};
+
+const HideSwipeOverlayCookieName = 'hide_swipe_overlay';
+
+export class SwipePrompt extends React.Component {
+  static propTypes = {
+    useMinecraftStyling: PropTypes.bool,
+
+    // from redux
+    buttonsAreVisible: PropTypes.bool.isRequired,
+    buttonsAreDisabled: PropTypes.bool.isRequired,
+    hasBeenDismissed: PropTypes.bool.isRequired,
+    onDismiss: PropTypes.func.isRequired,
+    dismissAction: PropTypes.string.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+    if (this.swipeOverlayOverrideSet()) {
+      cookies.remove(HideSwipeOverlayCookieName, {path: '/'});
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const {hasBeenDismissed, dismissAction} = this.props;
+    if (
+      hasBeenDismissed &&
+      !prevProps.hasBeenDismissed &&
+      !this.hideOverlayCookieSet()
+    ) {
+      // The overlay was just dismissed. Don't show it again for a while.
```

Comment:
How was this time threshold chosen?

Reply:
Some details

Our usage: There's another place we use cookies to delay re-displaying a helper prompt if a person has interacted with it. https://github.com/code-dot-org/code-dot-org/pull/34390 - The hide toolbar helper is re-displayed one year after it is dismissed.

Usage on mobile games (comparative study): many mobile games have a tutorial when a user first downloads the game. These walk through the interactions that are required in order to play the game. The user has to successfully complete the tutorial to continue and play the "full" version of the game. This pattern is followed in Clash of Clans, TempleRun2, and Mario Kart. 

Usage on console games (comparative study) - this is mostly from personal memory: Many console games have more complicated combinations of button presses. There is a tutorial at the beginning of these games. Additionally, as opportunities to use these button presses occur, they are displayed again on the screen (sometimes every time). This is pattern is followed in many first person shooter games.

Some details about swipe and this overlay:
- Swipe is significantly easier to use than the on-screen buttons. It's a significant improvement in user experience if we can teach touch-screen users to use swipe on our labs.
- Not all labs support swipe. Users might un-learn swipe if they switch between labs that do and don't support swipe.
- We don't have a "tutorial" for our labs. Users learn to interact with the labs slowly over the whole time of a course.
- (we have been considering creating a 'mobile tutorial,` but aren't there yet)
- The overlay is _very_ easy to dismiss. You can dismiss it by hitting an arrow key, an arrow button, or touching the screen.

My thoughts after testing these out: We don't have a metric that says 30 days is the appropriate number of days, but it's a starting point. Since it's easy to dismiss and is a significant improvement to experience, 30 days doesn't seem like _too_ often. We also don't want it showing up all the time - it could quickly get annoying if a user knows to use swipe.